spec/services/request_password_reset_spec.rb

```diff
@@ -86,5 +86,22 @@
         RequestPasswordReset.new(unconfirmed_email_address.email).perform
       end
     end
+
+    context 'when two users have the same email address' do
```

Comment:
I copied this test into `master` and it passed multiple times in a row. I then added a loop to run the test 10 times, and often got no failures. At 50 times, it passed the first time so I bumped to 100 times and started consistently getting failures.

```
      it 'always finds the user with the confirmed email address' do
        100.times do  |i|
          form = RequestPasswordReset.new(email)
          form.perform

          expect(form.send(:user)).to eq(@user_confirmed)
        end
      end
```

Reply:
Ok thanks --- mine consistently failed with one attempt.... 

Maybe for the duration of the test I could add in a default scope on the model for order by id and then take it off?

Reply:
@solipet take a look at 13ac53790e825d37561b3ca14c9b765674836f8a, does that fail more consistently for you?